### PR TITLE
[CBRD-23898] fix cppcheck errors other than broker and win_tools

### PR DESCRIPTION
**Pull request by cristiarg**
http://jira.cubrid.org/browse/CBRD-23898

Fix errors reported by cppcheck 1.9.

Remarks:
- not all errors have been fixed, there are still the following which need further actions, either in the context of this PR or separate:
- I did not know how to actually fix this, seems like a false negative because a `lseek` is actually performed on the associated file descriptor `vol_fd`:
```
src/storage/file_io.c:1519:14: error: Read and write operations without a call to a positioning function (fseek, fsetpos or rewind) or fflush in between result in undefined behaviour. [IOWithoutPositioning]
      (void) fprintf (fp, "%-10d %s %d %s %ld", *last_deleted_arv_num, login_name, (int) GETPID (), host, time (NULL));
             ^
```
- this one is a false negative that I was not able to get rid of:
```
unit_tests/memory_alloc/test_memory_alloc_helper.hpp:80:8: error: The deallocation of an auto-variable results in undefined behaviour. You should only free memory that has been allocated dynamically. [autovarInvalidDeallocation]
 free (pointer);
       ^
```

Question:
I have also run cppcheck 2.4 against the codebase and it uncovered some more issues - see comment on http://jira.cubrid.org/browse/CBRD-23908 .
Let me know if I should also fix the issues reported by 2.4 .

### CODE REVIEW

File: unit_tests/memory_alloc/test_db_private_alloc.cpp
Comment (cristiarg) [inline unit_tests/memory_alloc/test_db_private_alloc.cpp:255]: While cppcheck reported some error, it was not the actuall error.
This is the error that cppcheck 1.9 reported:
```
unit_tests/memory_alloc/test_db_private_alloc.cpp:255:15: error: The deallocation of an auto-variable results in undefined behaviour. You should only free memory that has been allocated dynamically. [autovarInvalidDeallocation]
    delete [] pointers;
              ^
```

With cppcheck 2.4, the error is no longer reported.

**Comment by wrlawodms** [comment]
We've upgraded the cppcheck version to 2.4.1. Please correct errors based on it. When it comes to some false positive errors, I'll take a look at them and let you know how to deal with them. We might suppress some error codes or ignore them.

**Comment by cristiarg** [comment]
> We've upgraded the cppcheck version to 2.4.1. Please correct errors based on it. When it comes to some false positive errors, I'll take a look at them and let you know how to deal with them. We might suppress some error codes or ignore them.

Ok, will work on it.

**Comment by cristiarg** [comment]
@wrlawodms
PR is ready for review.
Fixed errors reported by cppcheck 2.4.1

File: src/storage/extendible_hash.c
Comment (wrlawodms) [reply src/storage/extendible_hash.c:3415]: OK

**Review by wrlawodms — COMMENTED** [review]
I've reviewed `storage/, thread/, transaction/` and `unit_tests/`. @hgryoo will review the rest.

**Review by hgryoo — APPROVED** [review]
For the changes in src/cci/cci_query_execute.c, Could you initialize the variables (_data_) as C style like the following?
```
        INT64 data = 0LL;
	T_CCI_DATE data = { 0, 0, 0, 0, 0, 0, 0 };
        T_CCI_DATE_TZ data = { 0, 0, 0, 0, 0, 0, 0, ""};
```
As far as I know, CCI is a C Interface (driver) and the code is maintained as pure C. Additionally, CCI is going to be extracted as a separate repository (please refer, http://jira.cubrid.org/browse/CBRD-23840). So it would be okay to keep the code convention as it is.

The rest is perfect to me. :) Thank you for your contribution!

File: unit_tests/string_buffer/src/test_string_buffer.cpp
Comment (cristiarg) [reply unit_tests/string_buffer/src/test_string_buffer.cpp:107]: Provided logging.
The entire test `test_string_buffer` seems to fail anyway. A new issue should be submitted.

File: src/base/tz_support.c
Comment (cristiarg) [reply src/base/tz_support.c:1895]: Done

File: src/executables/esql_symbol_table.c
Comment (cristiarg) [reply src/executables/esql_symbol_table.c:752]: Done

File: src/storage/file_io.c
Comment (cristiarg) [reply src/storage/file_io.c]: I think it is a cppcheck false negative.
I replaced however and the error is not longer reported.

File: src/transaction/log_applier_sql_log.c
Comment (cristiarg) [reply src/transaction/log_applier_sql_log.c:141]: I think you are right.
I did not look at the context.
The function `sl_read_catalog` is used by `sl_init` as a confirmation step that the catalog has been written.

File: src/storage/file_io.c
Comment (wrlawodms) [reply src/storage/file_io.c]: Yes, I agree with you. What I suggested is just how to deal with the false negative. Thank you!

File: unit_tests/string_buffer/src/test_string_buffer.cpp
Comment (wrlawodms) [reply unit_tests/string_buffer/src/test_string_buffer.cpp:107]: I've run some unit tests, and it turns out some of them even raise core. We'll discuss it. Thank you.

**Review by wrlawodms — APPROVED** [review]

File: unit_tests/string_buffer/src/test_string_buffer.cpp
Comment (wrlawodms) [reply unit_tests/string_buffer/src/test_string_buffer.cpp:107]: We've decided to let it untouched. These unit tests are not managed now, and it is not easy for us to revise now. We could consider these unit tests later. Do you have an opinion on it?

Comment (cristiarg) [reply unit_tests/string_buffer/src/test_string_buffer.cpp:107]: I agree as I think they are out of the scope of the cppcheck related changes.
A new issue should be submitted that either these unit tests should be revived (preferred) or removed (not-preferred).

Comment (wrlawodms) [reply unit_tests/string_buffer/src/test_string_buffer.cpp:107]: Ok, then, could you submit the issue if you don’t mind? Let’s leave it to the maintainer to decide when to address it.

Comment (cristiarg) [reply unit_tests/string_buffer/src/test_string_buffer.cpp:107]: Added http://jira.cubrid.org/browse/CBRD-23923

**Pull request merged**
